sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1647,4 +1647,47 @@ private[spark] object QueryCompilationErrors {
   def invalidYearMonthIntervalType(startFieldName: String, endFieldName: String): Throwable = {
     new AnalysisException(s"'interval $startFieldName to $endFieldName' is invalid.")
   }
+
+  def configRemovedInVersion(configName: String, version: String, comment: String): Throwable = {
+    new AnalysisException(
+      s"The SQL config '$configName' was removed in the version $version. $comment")
+  }
+
+  def failedFallbackParsingError(msg: String, e1: Throwable, e2: Throwable): Throwable = {
+    new AnalysisException(s"$msg${e1.getMessage}\nFailed fallback parsing: ${e2.getMessage}",
+      cause = Some(e1.getCause))
+  }
+
+  def decimalCannotGreaterThanPrecisionError(scale: Int, precision: Int): Throwable = {
+    new AnalysisException(s"Decimal scale ($scale) cannot be greater than precision ($precision).")
+  }
+
+  def decimalOnlySupportPrecisionUptoError(decimalType: String, precision: Int): Throwable = {
+    new AnalysisException(s"$decimalType can only support precision up to $precision")
+  }
+
+  def negativeScaleNotAllowedError(scale: Int): Throwable = {
+    new AnalysisException(s"Negative scale is not allowed: $scale. " +
+      s"You can use spark.sql.legacy.allowNegativeScaleOfDecimal=true " +
+      s"to enable legacy mode to allow it.")
```

Comment:
Please refer https://github.com/apache/spark/blob/782dd13d720d0cc3a4799312920c5756ae5e7494/sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala#L986

Reply:
done, do you mean change spark.sql.legacy.allowNegativeScaleOfDecimal to ${LEGACY_ALLOW_NEGATIVE_SCALE_OF_DECIMAL_ENABLED.key}

Reply:
```
"""
  | ...
  |...
""".stripMargin.replaceAll("\n", " ")
```